keras/src/utils/saved_model_utils.py

```diff
@@ -0,0 +1,70 @@
+import json
+import zipfile
+
+import h5py
+
+from keras.saving import deserialize_keras_object
+from keras.src.saving.saving_lib import H5IOStore
+
+_CONFIG_FILENAME = "config.json"
+_METADATA_FILENAME = "metadata.json"
+_VARS_FNAME = "model.weights"
+
+
+class KerasFileEditor:
+    def __init__(self, filepath, reference_model=None, custom_objects=None):
+        self.filepath = filepath
+        self.custom_objects = custom_objects
+        self.metadata = None
+        self.reference_model = None
+        self.config = None
+
+        if filepath.endswith(".keras"):
+            self.init_for_keras(custom_objects, filepath, reference_model)
+        elif filepath.endswith(".weights.h5"):
+            pass
```

Comment:
Is this a TODO?

Reply:
I just added the logic to read weieghts.h5 file.